basics/getting_started/plugin_compatibility.md

```diff
@@ -2,80 +2,102 @@
 title: Plugin Compatibility with IntelliJ Platform Products
 ---
 
-<!--TODO link to sample_plugin file-->
+<style>
+  table {
+    width:100%;
+  }
+  th:first-child, td:first-child {
+    width: 30%;
+  }
+  th:last-child, td:last-child {
+    width: 25%;
+  }
+</style>
+
+## Introduction
+All products based on the _IntelliJ Platform_ are built on the same underlying API. 
+Some of these products share features built on top of the platform, such as Java support in IntelliJ IDEA and Android Studio. 
+Underlying those shared features are shared components. 
+When authoring a plugin for the IntelliJ Platform, it is important to understand and declare dependencies on these components. 
+Otherwise, it may not be possible to load or run the plugin in a product because the components on which it depends aren't available.
+
+## Declaring Plugin Dependencies
+For the purposes of dependencies, a _module_ can be thought of like a built-in plugin that ships as a non-removable part of a product. 
+A working definition of a dependency is that a plugin project cannot be run without the module present in an IntelliJ Platform-based product. 
+Declaring a dependency on a module also expresses a plugin's compatibility with a product in that the IntelliJ Platform determines whether a product contains the correct modules to support a plugin before loading it. 
+
+Note:
+* If a plugin does not include any module dependency tags in its `plugin.xml` file, it's assumed to be a legacy plugin and is loaded only in IntelliJ IDEA.
+  This form of dependency "non-declaration" is deprecated; do not use it for new plugin projects. 
+* If the `plugin.xml` file includes one or more dependency tags, the plugin is only loaded by the IntelliJ Platform if the product contains _all the modules_ on which the plugin depends.
+
+[Part I](/basics/plugin_structure/plugin_dependencies.md) of this document describes the syntax for declaring plugin dependencies and optional plugin dependencies.  
+Part II of this document (below) describes the functionality of the IntelliJ Platform modules to aid in determining the dependencies of a plugin. 
+
+## Modules
+A _module_ represents a built-in plugin that is a non-removable part of a product. 
+Some modules are available in all products, and some modules are available only in some, or even just one product. 
+This section identifies and discusses modules of both types.
+
+### Modules Available in All Products
+A core set of modules are available in all products based on the IntelliJ Platform. 
+These modules provide a set of shared functionality. 
+
+The following table lists modules that are currently available in all products. 
```

Comment:
It's worth to mention here that the list is not complete, there are other modules currently available in our IDEs which are not listed here.

Reply:
I have added a note to that effect, but I'd like to give readers a way to find them. Is it just searching plugin.xml files for lines containing the string '<module value="'?

Reply:
Most of the modules are declared not in plugin.xml files in other files included into the main "platform" *Plugin.xml file, e.g. JavaPlugin.xml, PlatformLangPlugin.xml etc.